### PR TITLE
Fixed some spacing issues with paragraphs and headings

### PR DESCRIPTION
**Pull request by eskirk**
Some naming conventions were changed so @rnetro and I went through some of the CSS to normalize names and spacings and other good shtuff ( ° ͜ʖ °)

### CODE REVIEW

File: src/notebook/index.css [hunk 1]
```diff
@@ -9,6 +9,11 @@
 }
 
 
+.p-Widget * + p {
```
Comment (sccolbert) [inline src/notebook/index.css]: This is a pretty broad selector (and expensive to boot). What are wanting to accomplish here? There might be a better way to select it, or we can add an appropriate class name to make it easier to target.

Comment (eskirk) [inline src/notebook/index.css]: We were running into an issue where there wouldn't be any spacing after a heading when there was a cell that had multiple p and h elements within it. I was thinking of some kind of workaround involving class names but I didn't want to go that deep into this issue.

The only reason we did it this way was that when we inspected the Notebook in chrome to try and match the style, we saw pretty much that exact line of code, tried it out, and it worked. If it is too expensive to use then we can take it out, it really doesn't make a difference. 

I could make a rule like `h1 + p { /* rule stuff */ }` and apply that for each header, but that would be ugly and I don't know enough about CSS to do something that nasty.

Comment (sccolbert) [inline src/notebook/index.css]: I think we should remove this rule from this PR and open a separate issue about it with a full explanation. Then, we can design the document structure and class names (if needed) to make it easy to select and style. The rest of this PR seems reasonable.

File: src/notebook/index.css [hunk 2]
```diff
@@ -37,67 +42,109 @@
...
-.jp-MarkdownCell-renderer.p-Widget h2 {
+.jp-MarkdownCell-content.p-Widget h2 {
   font-family: "Helvetica Neue", Arial, Helvetica, Geneva, sans-serif;
   font-size: 22px;
   line-height: 1.0;
+  -webkit-margin-after: 0em;
+  -webkit-margin-before: 0em;
+  -webkit-margin-start: 0em;
+  -webkit-margin-end: 0em;
+  margin-top: 14px;
```
Comment (sccolbert) [inline src/notebook/index.css]: seems like we can do without the `-webkit-` prefixes if we explicitly specify the margins:

`margin: 14px 0 0 0`

http://stackoverflow.com/questions/5721966/webkit-margin-adds-unwanted-margin-on-texts

Comment (eskirk) [inline src/notebook/index.css]: The webkit-margin stuff was being inherited from Chrome's style sheet and was adding some unwanted spacing. We asked over Gitter and were given advice to do it this way, is there another way to reset those margins?

Comment (eskirk) [inline src/notebook/index.css]: Will that overwrite Chrome's style sheet's margins? That was the problem we were trying to fix.

Comment (sccolbert) [inline src/notebook/index.css]: according to that stack overflow, yes. 

I would be surprised if it didnt.

Comment (eskirk) [inline src/notebook/index.css]: I have been trying to fix these margin problems with `-webkit-margin` but just setting `margin` isn't really helping the problem. Any tips here?

**Comment by sccolbert** [comment]
closing in favor of #137

**Pull request closed without merging**